src/coreComponents/dataRepository/xmlWrapper.cpp

```diff
@@ -169,6 +195,207 @@ string buildMultipleInputXML( string_array const & inputFileList,
   return inputFileName;
 }
 
+bool isFileMetadataAttribute( string const & name )
+{
+  static const std::set< string > fileMetadataAttributes {
+    "name", "xmlns:xsi", "xsi:noNamespaceSchemaLocation", xmlWrapper::filePathString, xmlWrapper::charOffsetString
+  };
+  return fileMetadataAttributes.find( name ) != fileMetadataAttributes.end();
+}
+
+const size_t xmlDocument::npos = string::npos;
+size_t documentId=0;
+
+xmlDocument::xmlDocument():
+  pugi::xml_document(),
+  m_rootFilePath( "CodeIncludedXML" + std::to_string( documentId++ ) )
+{}
+
+xmlResult xmlDocument::load_string( const pugi::char_t * contents, bool loadNodeFileInfo,
+                                    unsigned int options )
+{
+  xmlResult result = pugi::xml_document::load_string( contents, options );
+
+  // keeping a copy of original buffer to allow line retrieval
+  if( loadNodeFileInfo )
+  {
+    new (&m_originalBuffers) map< string, string >();
+    m_originalBuffers[m_rootFilePath] = string( contents );
```

Comment:
Because the `xmlDocument::load_*` methods destroy the current xml tree. I need to free the previously loaded buffers to keep being synchronized with the loaded xml tree.

I'll add *"Free the existing tree then load ..."* to the comments of `xmlDocument::load_*` methods.

Reply:
I mistakenly thought that `std::map::clear()` does not clear all used memory, since it worked that way for `std::unordered_map::clear()`, and I forgot that the `new` syntax doesn't call the destructor.
I'll use `clear()` here.